lib/state_machine.js

```diff
@@ -189,7 +200,15 @@ limitations under the License.
         // Whoa! Some kind of exception occurred while we were attempting to do
         // a transition. Delegate to the error handler. The error handler is
         // responsible for resuming the state machine.
-        this.handleError(error);
+        if (this.errorListener_) {
+          this.errorListener_(error);
+        }
+        
+        // Go back to the initial state.
+        this.transitionTo(this.initialState_);
```

Comment:
...if your state is initialState_ is it bad to transition back to it?  Just worried about looping...

Reply:
This is a valid concern in general, but in this case, IdleState (the common initial state) is always safe to go back to.